src/Core/src/Platform/iOS/WrapperView.cs

```diff
@@ -119,9 +120,37 @@ public override CGSize SizeThatFits(CGSize size)
 
 			var child = Subviews[0];
 
+			// Calling SizeThatFits on an ImageView always returns the image's dimensions, so we need to call the extension method
+			// This also affects ImageButtons
+			if (child is UIImageView imageView)
+			{
+				return imageView.SizeThatFitsImage(size);
+			}
+			else if (child is UIButton imageButton && imageButton.ImageView?.Image is not null)
+			{
+				return imageButton.ImageView.SizeThatFitsImage(size);
+			}
+
 			return child.SizeThatFits(size);
 		}
 
+		internal CGSize SizeThatFitsWrapper(CGSize originalSpec, double virtualViewWidth, double virtualViewHeight)
+		{
+			if (Subviews.Length == 0)
+				return base.SizeThatFits(originalSpec);
+
+			var child = Subviews[0];
+
+			if (child is UIImageView || (child is UIButton imageButton && imageButton.ImageView?.Image is not null))
+			{
+				var widthConstraint = IsExplicitSet(virtualViewWidth) ? virtualViewWidth : originalSpec.Width;
+				var heightConstraint = IsExplicitSet(virtualViewHeight) ? virtualViewHeight : originalSpec.Height;
+				return SizeThatFits(new CGSize(widthConstraint, heightConstraint));
+			}
```

Comment:
That's a good point. This was put to ensure that the `WrapperView` was doing the right thing. Looking back this can just go away and the logic be put inside the `WraperView`'s SizeThatFits override

Reply:
Oh wait now I remember why. I need the VirtualView's height and and width. I can only access that from ViewHandlerExtensions.iOS.cs

Reply:
While the argument can be made that I could just remove the WrapperView's `SizeThatFits` override and just use `SizeThatFitsWrapper` I was afraid of then breaking other parts of the code that may call this method. There are places where the WrapperView is cast as a standard `UIView`, so in order to possibly break something I kept it.